src/main/java/radiounit/EricssonLteRadioUnitReceiver.java

```diff
@@ -13,60 +15,193 @@
 	}
 
 	public void activateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] activateEricssonLte");
 	}
 
 	public void deactivateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] deactivateEricssonLte");
 	}
 
 	public void releaseEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] releaseEricssonLte");
 	}
 
-	public void setupCarrierEricssonLte(Carrier carrrier) {
-		// TODO Auto-generated method stub
-
+	public void setupCarrierEricssonLte(Carrier carrier) {
+		System.out.println("[EricssonLteRadioUnitReceiver] setupCarrierEricssonLte: " + carrier);
+		
+		// Validate carrier
+		FrequencyBand band = carrier.getCarrierFrequencies();
+		boolean isLte = false;
+		
+		for (LteFrequencyBand lteFreq : LteFrequencyBand.values())
+		{
+			if (band == lteFreq)
+			{
+				isLte = true;
+				break;
+			}
+		}
+
+		if (!isLte)
+		{
+			System.err.println("Cannot add WCDMA carrier to LTE radio!");
+			return;
+		}
+		
+		if (getCarrierById(carrier.getCarrierId()) != null)
+		{
+			System.err.println("Carrier with id " + carrier.getCarrierId() + " already exists!");
+			return;
+		}
+		
+		addCarrierInternal(carrier);
 	}
 
 	public void signalScalingEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] signalScalingEricssonLte");
 	}
 
 	public void modifyCarrierEricssonLte(Integer carrierId, FrequencyBand frequencyBand) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] modifyCarrierEricssonLte: " + carrierId + ", " + frequencyBand.getBand());
+		
+		int idx = getCarrierIdxById(carrierId);
+		
+		if (idx == -1)
+		{
+			System.err.println("Carrier with id " + carrierId + " does not exist!");
+			return;
+		}
+		
+		Carrier existingCarrier = getCarrierById(carrierId);
+		
+		removeCarrierInternal(idx);
+		
+		existingCarrier.setFrequencyBand(frequencyBand);
+		
+		addCarrierInternal(existingCarrier);
 	}
 
 	public void removeCarrierEricssonLte(Integer carrierId) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] removeCarrierEricssonLte: " + carrierId);
+		
+		int idx = getCarrierIdxById(carrierId);
+		
+		if (idx == -1)
+		{
+			System.err.println("Carrier with id " + carrierId + " does not exist!");
+			return;
+		}
+		
+		removeCarrierInternal(idx);
 	}
 
 	public void selfDiagnosticsEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] selfDiagnosticsEricssonLte");
 	}
 
-	public void removeAllCharactersEricssonLte() {
-		// TODO Auto-generated method stub
-
+	public void removeAllCarriersEricssonLte() {
+		System.out.println("[EricssonLteRadioUnitReceiver] removeAllCarriersEricssonLte");
+		
+		for (int i = 0; i < getAllCarriersInternal().size(); ++i)
```

Comment:
Minor: Oh, I'm curious why you used `++i` rather than `i++`. If there wasn't any particular reason, that's okay too.

Reply:
Functionally, they are the same - [this SO explains the difference](https://stackoverflow.com/questions/2315705/what-is-the-difference-between-i-i-in-a-for-loop).

I'm not 100% on if it's in the same in Java - but for C++ performance is *slightly* better with prefix (++i) rather than postfix (i++), since postfix requires copying to a temporary variable, but prefix does not. It could be that Java optimizes this anyway, but I'm just using what I'm used to :)

Reply:
Ah, okay, thanks for the information.